crates/fuel-core/src/executor.rs

```diff
@@ -737,8 +761,12 @@ impl Executor {
                 Input::Contract { .. } => {}
                 Input::MessageSigned { message_id, .. }
                 | Input::MessagePredicate { message_id, .. } => {
-                    if let Some(message) = db.storage::<Messages>().get(message_id)? {
-                        if message.fuel_block_spend.is_some() {
+                    if let Some(message) = self
```

Comment:
We also need to verify that all fields of the message from the predicate are the same in the relayer.

It seems we had a bug previously, so we can create a separate issue for that=)